### ISSUE TITLE
[Bug]: Error occurs when query `information_schema`.`tables`

### ISSUE THREAD

**Issue by DanielZhangQD** [matrixorigin/matrixone#15332]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

1.1.2

### Commit ID

v1.1.2-1519f8e5c-2024-04-02

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

```
MySQL [(none)]> select * from information_schema.tables;
ERROR 20301 (HY000): invalid input: Table 'statement_cu' does not exist in database 'mo_catalog'
```

### Expected Behavior

Query succeeds

### Steps to Reproduce

```Markdown
Execute the sql in cloud instance
```

**Comment by DanielZhangQD** [comment]
'statement_cu' is a cluster table, not sure if it's related to the MO upgrade.

**Comment by sukki37** [comment]
If use sys account to execute the same sql, get another error.
```
mysql> select * from information_schema.tables;
ERROR 20301 (HY000): invalid input: Database 'api_iott' does not exist
```

**Comment by qingxinhome** [comment]
The issue has been resolved and PR has been submitted to main and 1.1 dev. Please verify and test. @DanielZhangQD @sukki37 @daviszhen

**Comment by aressu1985** [comment]
done
